src/main/java/io/nats/client/impl/NatsSubscription.java

```diff
@@ -68,11 +68,11 @@ public boolean isActive() {
         return (this.dispatcher != null || this.incoming != null);
     }
 
-    void setBeforeQueueProcessor(Function<NatsMessage, NatsMessage> beforeQueueProcessor) {
-        this.beforeQueueProcessor = beforeQueueProcessor; // better not be null if it's being set
+    void setBeforeQueueProcessor(Function<NatsMessage, Boolean> beforeQueueProcessor) {
```

Comment:
does this need to be synchronized guarded?

Reply:
No. This is used internally during construction or post construction